message_ix_models/model/water/data/water_for_ppl.py

```diff
@@ -171,6 +177,91 @@ def hist_cap(x: pd.Series, context: "Context", hold_cost: pd.DataFrame) -> list:
     ]
 
 
+def relax_growth_constraint(ref_hist, scen, cooling_df, g_lo, constraint_type):
```

Comment:
This function could really use some type hints! In particular, if you type hint `constraint_type: Literal[Union["activity", "capacity"]]` (as a [Literal](https://docs.python.org/3/library/typing.html#typing.Literal), which you might have to import from typing first (maybe `Union`, too)), you can save yourself from manually having to raise a `ValueError`: the function will automatically fail if people try to pass in invalid strings!